src/buddy/reporting/XUnit2Reporter.hx

```diff
@@ -0,0 +1,221 @@
+package buddy.reporting;
+
+import sys.io.File;
+import haxe.CallStack.StackItem;
+import buddy.BuddySuite.Spec;
+import buddy.BuddySuite.Suite;
+import buddy.reporting.Reporter;
+import promhx.Deferred;
+import promhx.Promise;
+
+using StringTools;
+
+class XUnit2Reporter implements Reporter {
+    var reportName : String;
+
+    var total : Int;
+
+    var passing : Int;
+
+    var failures : Int;
+
+    var pending : Int;
+
+    var unknowns : Int;
+
+    var xml : Xml;
+
+    var totalTime : Float;
+
+    public function new() {
+        if (!isDefined('report-name')) {
+            throw 'report-name not defined';
+        }
+
+        reportName = if (isDefined('report-name')) getDefine('report-name') else 'report';
+        total      = 0;
+        passing    = 0;
+        failures   = 0;
+        pending    = 0;
+        unknowns   = 0;
+        totalTime  = 0;
+
+        xml = Xml.createElement('assemblies');
+        xml.set('timestamp', '${getDate()} ${getTime()}');
+    }
+
+    static macro function isDefined(key : String) : haxe.macro.Expr {
```

Comment:
Why do you use macro functions instead of `#if` / `#error`?

Reply:
I just used the code from the haxe cookbook page on checking and getting custom compiler flags, is there a better way than a macro function?

Reply:
I'd say it's fine to use it as an expression here, since it's more about basic settings than including/excluding functionality. But according to line 31, it must always be defined?

Reply:
Ah, yes. I used to have it so that it always needed to be defined but swapped it to being optional for this PR, but I guess I forgot to remove the old check. Should be fixed now!